R/utils.R

```diff
@@ -98,22 +98,40 @@ as_binary <- function(x) {
   as.integer(x)
 }
 
-random_string <- function(size = NULL) {
-  # creates a random string using tempfile() which does better to not affect the
-  # random seed
+#' random string function that does not alter the seed.
+#' 
+#' simple wrapper around `stringi::stri_rand_strings()``
+#' 
+#' @param n n
+#' @param length length
+#' @param pattern pattern
+#' @keywords internal
+#' @noRd
+random_string <- function(n = 1, length, pattern = "[A-Za-z0-9]") {
   # https://github.com/ycphs/openxlsx/issues/183
   # https://github.com/ycphs/openxlsx/pull/224
-  res <- basename(tempfile(""))
+  
+  if (missing(length)) length <- 16
```

Comment:
Just set the default value to `16`?

Reply:
eh, haha, yes. I kinda assumed that length was not required for stringi, fixed it but to tired :)